invenio_rdm_pure/source/rdm/run/synchronizer.py

```diff
@@ -12,67 +12,84 @@
 import time
 import traceback
 from concurrent.futures import ThreadPoolExecutor
+from os.path import dirname, isabs, isfile, join
+from pathlib import Path
 from typing import List
 
 from flask import current_app
+from flask_principal import Identity
+from invenio_access.permissions import any_user
+from invenio_pidstore.resolver import Resolver
+from invenio_records_marc21.services import Marc21RecordService, Metadata, RecordItem
 
 from ...pure.requests_pure import (
-    get_pure_metadata,
+    download_pure_file,
     get_research_output_count,
     get_research_outputs,
 )
 from ...utils import get_dates_in_span
-from ..converter import Converter
+from ..converter import Converter, Marc21Record
+from ..database import RdmDatabase
 
 
 class Synchronizer(object):
     """Synchronizer class to facilitate record synchronization between Invenio and Pure."""
 
-    def __init__(self):
-        """Default Constructor of the class Synchronizer."""
+    def __init__(
+        self,
+        pure_api_url: str = "",
+        pure_api_key: str = "",
+        pure_username: str = "",
+        pure_password: str = "",
+    ):
+        """Default Constructor of the Synchronizer class."""
+        if not pure_api_url:
```

Comment:
it would be nicer to have the following statements in the lines 40 - 43. so not to have there empty strings but the real default value

Reply:
I don't think it would work and would be good practice. Default method arguments should be compile-time constants.

Reply:
a function call is possible. I didn't kew compile-time constants exists in python. But it is ok, let it as it is

Reply:
Otherwise it throws a `RuntimeError: Working outside of application context.` error.

Reply:
ahhhh, because of the current_app part. then let it as it is.